src/mqt/bench/devices/ibm.py

```diff
@@ -10,136 +10,58 @@
 
 from __future__ import annotations
 
-import json
-from typing import TYPE_CHECKING, TypedDict, cast
-
-if TYPE_CHECKING:
-    from pathlib import Path
-
-    from qiskit.providers import BackendV2
-    from qiskit.transpiler import Target
+import logging
+from typing import TYPE_CHECKING, cast
 
+from qiskit_ibm_runtime.fake_provider import FakeMontrealV2, FakeTorino, FakeWashingtonV2
 
+from . import Gateset
 from .calibration import DeviceCalibration
 from .device import Device
-from .provider import Provider
-
-
-class QubitProperties(TypedDict):
-    """Class to store the properties of a single qubit."""
-
-    T1: float  # us
-    T2: float  # us
-    eRO: float
-    tRO: float  # ns
-    eID: float
-    eSX: float
-    eX: float
-    eCX: dict[str, float]
-    tCX: dict[str, float]  # ns
-
-
-class IBMCalibration(TypedDict):
-    """Class to store the calibration data of an IBM device."""
-
-    name: str
-    basis_gates: list[str]
-    num_qubits: int
-    connectivity: list[list[int]]
-    properties: dict[str, QubitProperties]
-
-
-class IBMProvider(Provider):
-    """Class to manage IBM devices."""
-
-    provider_name = "ibm"
 
-    @classmethod
-    def get_available_device_names(cls) -> list[str]:
-        """Get the names of all available IBM devices."""
-        return ["ibm_washington", "ibm_montreal"]  # NOTE: update when adding new devices
+logger = logging.getLogger(__name__)
 
-    @classmethod
-    def get_native_gates(cls) -> list[str]:
-        """Get a list of provider specific native gates."""
-        return ["id", "rz", "sx", "x", "cx", "measure", "barrier"]  # washington, montreal
-
-    @classmethod
-    def import_backend(cls, path: Path) -> Device:
-        """Import an IBM backend.
+if TYPE_CHECKING:
+    from qiskit.providers import BackendV2
 
-        Arguments:
-            path: the path to the JSON file containing the calibration data.
 
-        Returns: the Device object
-        """
-        with path.open() as json_file:
-            ibm_calibration = cast("IBMCalibration", json.load(json_file))
+class IBMDevice(Device):
+    """IBM Washington device."""
 
-        device = Device()
-        device.name = ibm_calibration["name"]
-        device.num_qubits = ibm_calibration["num_qubits"]
-        device.basis_gates = ibm_calibration["basis_gates"]
-        device.coupling_map = list(ibm_calibration["connectivity"])
+    def __init__(self, backend: BackendV2, device_name: str, gateset_name: str) -> None:
+        """Initialize the IBM device."""
+        self.name = device_name
+        self.gateset = Gateset(gateset_name, backend.operation_names)
+        self.num_qubits = backend.num_qubits
+        self.coupling_map = backend.coupling_map
+        self.calibration = None
+        self.target = backend.target
```

Comment:
That's probably the best sign that the device class itself is (or at least many of its members are) redundant. We already store the target here and that contains practically all information.